recipe/meta.yaml

```diff
@@ -1,45 +1,151 @@
+{% set name = "recbole" %}
+{% set version = "1.2.0" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: <sha256>
+  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/recbole-{{ version }}.tar.gz
+    sha256: 132a8a706462b26bed82d409b67fbc09e90b3a49b607c7ba76c2f1718d8ddc17
+    patches:
+      - patches/0001-update-colorlog-pinning.patch
+      - patches/0002-update-randomstate-to-default_rng-for-hyperopt.patch
+  # use github release archive for tests
+  - url: https://github.com/RUCAIBox/RecBole/archive/refs/tags/v{{ version }}.tar.gz
+    sha256: 1bc77a567083884b41ddd652b9b35078a0c3554c39ca38ab825ab2681eef138e
+    folder: gh_src
 
 build:
   number: 0
+  # s390x: kmeans-pytorch (numba) unavailable
+  # py>=312: ray unavailable
+  skip: True  # [py<38 or py>311 or s390x]
   script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
-  skip: True  # [py<38]
 
 requirements:
+  build:
+    - patch     # [not win]
+    - m2-patch  # [win]
   host:
     - python
     - pip
+    - setuptools
+    - wheel
   run:
     - python
+    - pytorch >=1.10.0
+    - numpy >=1.17.2
+    # Introduced upper pinning for scipy because of
+    # https://github.com/RUCAIBox/RecBole/issues/2034
+    # https://github.com/RUCAIBox/RecBole/issues/2090
+    # similar to any other like
+    # https://github.com/mmaelicke/scikit-gstat/issues/178
+    # it can be removed when issues are closed.
+    - scipy >=1.6.0,<1.13.0a0
+    - pandas >=1.3.0
+    - tqdm >=4.48.2
+    # colorlog pinning relaxed to lower bound instead of exact
+    # (we have more recent version and it seems to work).
+    - colorlog >=4.7.2
+    - colorama 0.4.4
+    - scikit-learn >=0.23.2
+    - pyyaml >=5.1.0
+    - tensorboard >=2.5.0
+    - thop >=0.1.1.post2207130030
+    - tabulate >=0.8.10
+    - plotly >=4.0.0
+    - texttable >=0.9.0
+    - psutil >=5.9.0
+    # ray-tune is needed:
+    # https://github.com/RUCAIBox/RecBole/blob/v1.2.0/recbole/quick_start/quick_start.py#L20
+    # https://github.com/RUCAIBox/RecBole/blob/v1.2.0/conda/meta.yaml#L44
+    - ray-tune >=1.13.0,<=2.6.3
+    # The following dependencies are specified in the CI
+    # and it's not clear if they are mandatory or not.
+    # We contacted the upstream authors but received no answer,
+    # therefore we decided to set those dependencies as
+    # runtime dependencies and ship most of the recommenders
+    # except KGAT (missing 'dgl'), KGIN, and MCCLK (missing
+    # 'torch-scatter').
+    # https://github.com/RUCAIBox/RecBole/blob/v1.2.0/.github/workflows/python-package.yml#L36-L51
+    - kmeans-pytorch
+    - xgboost
+    - lightgbm
+    - faiss-cpu
+    - networkx
+    - protobuf #3.19.0
+    # originally ==0.2.5 adapted with a patch
+    # to work with the more recent version we
+    # have (0.2.7)
+    - hyperopt >=0.2.5
```

Comment:
should it be run_constrained? It's an extras_require

Reply:
it is,

the suggested smoke test from upstream uses it, though (see my comment in the issue here: https://github.com/RUCAIBox/RecBole/issues/2105).

unfortunately, I did not receive any answer also for the rest of the seemingly optional dependencies (see comment a few lines before, and also in the above issue link).

Therefore, I moved all of those optional deps except some that we do not have to `run`.

Reply:
recbole upstream replied that the following dependencies are optional:

torch-scatter, dgl and kmeans-pytorch are all optional.

ref: https://github.com/RUCAIBox/RecBole/issues/2105#issuecomment-2559262386

Therefore, I am moving the optional dependencies in `run_constrained`.